diff --git a/hudi-common/src/test/java/org/apache/hudi/common/table/read/SchemaHandlerTestBase.java b/hudi-common/src/test/java/org/apache/hudi/common/table/read/SchemaHandlerTestBase.java
index 41d63434f5d76..fe9423cdc5536 100644
--- a/hudi-common/src/test/java/org/apache/hudi/common/table/read/SchemaHandlerTestBase.java
+++ b/hudi-common/src/test/java/org/apache/hudi/common/table/read/SchemaHandlerTestBase.java
@@ -251,11 +251,11 @@ private static void setupMORTable(RecordMergeMode mergeMode, boolean hasPrecombi
     when(hoodieTableConfig.populateMetaFields()).thenReturn(true);
     when(hoodieTableConfig.getRecordMergeMode()).thenReturn(mergeMode);
     if (hasPrecombine) {
-      when(hoodieTableConfig.getPreCombineFieldsStr()).thenReturn(Option.of("timestamp"));
-      when(hoodieTableConfig.getPreCombineFields()).thenReturn(Collections.singletonList("timestamp"));
+      when(hoodieTableConfig.getOrderingFieldsStr()).thenReturn(Option.of("timestamp"));
+      when(hoodieTableConfig.getOrderingFields()).thenReturn(Collections.singletonList("timestamp"));
     } else {
-      when(hoodieTableConfig.getPreCombineFieldsStr()).thenReturn(Option.empty());
-      when(hoodieTableConfig.getPreCombineFields()).thenReturn(Collections.emptyList());
+      when(hoodieTableConfig.getOrderingFieldsStr()).thenReturn(Option.empty());
+      when(hoodieTableConfig.getOrderingFields()).thenReturn(Collections.emptyList());
     }
     if (mergeMode == CUSTOM) {
       when(hoodieTableConfig.getRecordMergeStrategyId()).thenReturn("asdf");
