diff --git a/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/OptionsResolver.java b/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/OptionsResolver.java
index aa453fe54cdb0..375c3de809a04 100644
--- a/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/OptionsResolver.java
+++ b/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/OptionsResolver.java
@@ -152,12 +152,12 @@ public static boolean isDefaultHoodieRecordPayloadClazz(Configuration conf) {
   }
 
   /**
-   * Returns the preCombine field
+   * Returns the preCombine fields as comma separated string
    * or null if the value is set as {@link FlinkOptions#NO_PRE_COMBINE}.
    */
-  public static String getPreCombineField(Configuration conf) {
-    final String preCombineField = conf.get(FlinkOptions.PRECOMBINE_FIELD);
-    return preCombineField.equals(FlinkOptions.NO_PRE_COMBINE) ? null : preCombineField;
+  public static String getOrderingFieldsStr(Configuration conf) {
+    final String orderingFields = conf.get(FlinkOptions.ORDERING_FIELDS);
+    return orderingFields.equals(FlinkOptions.NO_PRE_COMBINE) ? null : orderingFields;
   }
 
   /**
