diff --git a/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/TableOptionProperties.java b/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/TableOptionProperties.java
index de6b6bc4111f4..c7781171b9992 100644
--- a/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/TableOptionProperties.java
+++ b/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/TableOptionProperties.java
@@ -90,13 +90,13 @@ public class TableOptionProperties {
 
     KEY_MAPPING.put("type", FlinkOptions.TABLE_TYPE.key());
     KEY_MAPPING.put("primaryKey", FlinkOptions.RECORD_KEY_FIELD.key());
-    KEY_MAPPING.put("preCombineField", FlinkOptions.PRECOMBINE_FIELD.key());
+    KEY_MAPPING.put("preCombineField", FlinkOptions.ORDERING_FIELDS.key());
     KEY_MAPPING.put("payloadClass", FlinkOptions.PAYLOAD_CLASS_NAME.key());
     KEY_MAPPING.put(SPARK_SOURCE_PROVIDER, CONNECTOR.key());
     KEY_MAPPING.put(FlinkOptions.KEYGEN_CLASS_NAME.key(), FlinkOptions.KEYGEN_CLASS_NAME.key());
     KEY_MAPPING.put(FlinkOptions.TABLE_TYPE.key(), "type");
     KEY_MAPPING.put(FlinkOptions.RECORD_KEY_FIELD.key(), "primaryKey");
-    KEY_MAPPING.put(FlinkOptions.PRECOMBINE_FIELD.key(), "preCombineField");
+    KEY_MAPPING.put(FlinkOptions.ORDERING_FIELDS.key(), "preCombineField");
     KEY_MAPPING.put(FlinkOptions.PAYLOAD_CLASS_NAME.key(), "payloadClass");
   }
 
