diff --git a/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/CompactionUtil.java b/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/CompactionUtil.java
index 42df2ced0cdd4..4c482835d2035 100644
--- a/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/CompactionUtil.java
+++ b/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/CompactionUtil.java
@@ -100,10 +100,10 @@ public static void setAvroSchema(HoodieWriteConfig writeConfig, HoodieTableMetaC
    *
    * @param conf The configuration
    */
-  public static void setPreCombineField(Configuration conf, HoodieTableMetaClient metaClient) {
-    String preCombineField = metaClient.getTableConfig().getPreCombineFieldsStr().orElse(null);
-    if (preCombineField != null) {
-      conf.set(FlinkOptions.PRECOMBINE_FIELD, preCombineField);
+  public static void setOrderingFields(Configuration conf, HoodieTableMetaClient metaClient) {
+    String orderingFields = metaClient.getTableConfig().getOrderingFieldsStr().orElse(null);
+    if (orderingFields != null) {
+      conf.set(FlinkOptions.ORDERING_FIELDS, orderingFields);
     }
   }
 
