diff --git a/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala b/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala
index 54a3577f7f6c8..5cbfdcef050e4 100644
--- a/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala
+++ b/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala
@@ -136,7 +136,7 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
     }
 
   protected lazy val preCombineFields: List[String] = {
-    val tablePrecombineFields = tableConfig.getPreCombineFields
+    val tablePrecombineFields = tableConfig.getOrderingFields
     if (tablePrecombineFields.isEmpty) {
       DataSourceOptionsHelper.getPreCombineFields(optParams)
         .orElse(java.util.Collections.emptyList[String])
