diff --git a/hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/TestDefaultSparkRecordMerger.java b/hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/TestDefaultSparkRecordMerger.java
index ccdcd1878055d..eaa31036a8726 100644
--- a/hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/TestDefaultSparkRecordMerger.java
+++ b/hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/TestDefaultSparkRecordMerger.java
@@ -42,7 +42,7 @@
 import java.util.List;
 
 import static org.apache.hudi.common.model.HoodieRecord.HoodieRecordType.SPARK;
-import static org.apache.hudi.common.table.HoodieTableConfig.PRECOMBINE_FIELDS;
+import static org.apache.hudi.common.table.HoodieTableConfig.ORDERING_FIELDS;
 import static org.apache.hudi.common.testutils.HoodieTestDataGenerator.TRIP_EXAMPLE_SCHEMA;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
@@ -100,7 +100,7 @@ void testMergerWithNewRecordAccepted() throws IOException {
 
     DefaultSparkRecordMerger merger = new DefaultSparkRecordMerger();
     TypedProperties props = new TypedProperties();
-    props.setProperty(PRECOMBINE_FIELDS.key(), INT_COLUMN_NAME);
+    props.setProperty(ORDERING_FIELDS.key(), INT_COLUMN_NAME);
     Schema avroSchema = AvroConversionUtils.convertStructTypeToAvroSchema(
         SPARK_SCHEMA, ANY_NAME, ANY_NAMESPACE);
     Option<Pair<HoodieRecord, Schema>> merged =
@@ -127,7 +127,7 @@ void testMergerWithOldRecordAccepted() throws IOException {
 
     DefaultSparkRecordMerger merger = new DefaultSparkRecordMerger();
     TypedProperties props = new TypedProperties();
-    props.setProperty(PRECOMBINE_FIELDS.key(), INT_COLUMN_NAME);
+    props.setProperty(ORDERING_FIELDS.key(), INT_COLUMN_NAME);
     Schema avroSchema = AvroConversionUtils.convertStructTypeToAvroSchema(
         SPARK_SCHEMA, ANY_NAME, ANY_NAMESPACE);
     Option<Pair<HoodieRecord, Schema>> r =
@@ -151,7 +151,7 @@ void testMergerWithNewRecordAsDelete() throws IOException {
 
     DefaultSparkRecordMerger merger = new DefaultSparkRecordMerger();
     TypedProperties props = new TypedProperties();
-    props.setProperty(PRECOMBINE_FIELDS.key(), INT_COLUMN_NAME);
+    props.setProperty(ORDERING_FIELDS.key(), INT_COLUMN_NAME);
     Schema avroSchema = AvroConversionUtils.convertStructTypeToAvroSchema(
         SPARK_SCHEMA, ANY_NAME, ANY_NAMESPACE);
     Option<Pair<HoodieRecord, Schema>> r =
@@ -172,7 +172,7 @@ void testMergerWithOldRecordAsDelete() throws IOException {
 
     DefaultSparkRecordMerger merger = new DefaultSparkRecordMerger();
     TypedProperties props = new TypedProperties();
-    props.setProperty(PRECOMBINE_FIELDS.key(), INT_COLUMN_NAME);
+    props.setProperty(ORDERING_FIELDS.key(), INT_COLUMN_NAME);
     Schema avroSchema = AvroConversionUtils.convertStructTypeToAvroSchema(
         SPARK_SCHEMA, ANY_NAME, ANY_NAMESPACE);
     Option<Pair<HoodieRecord, Schema>> r =
