diff --git a/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkSqlWithCustomKeyGenerator.scala b/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkSqlWithCustomKeyGenerator.scala
index 712b66c84cbc1..5672d4e95f11d 100644
--- a/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkSqlWithCustomKeyGenerator.scala
+++ b/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkSqlWithCustomKeyGenerator.scala
@@ -625,7 +625,7 @@ class TestSparkSqlWithCustomKeyGenerator extends HoodieSparkSqlTestBase {
       .option("hoodie.datasource.write.keygenerator.class", keyGenClassName)
       .option("hoodie.datasource.write.partitionpath.field", writePartitionFields)
       .option("hoodie.datasource.write.recordkey.field", "id")
-      .option("hoodie.datasource.write.precombine.field", "name")
+      .option(HoodieTableConfig.ORDERING_FIELDS.key(), "name")
       .option("hoodie.table.name", tableName)
       .option("hoodie.insert.shuffle.parallelism", "1")
       .option("hoodie.upsert.shuffle.parallelism", "1")
