diff --git a/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/cdc/TestCDCStreamingSuite.scala b/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/cdc/TestCDCStreamingSuite.scala
index 4ba1eb76b739e..d39ba79c9f0db 100644
--- a/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/cdc/TestCDCStreamingSuite.scala
+++ b/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/cdc/TestCDCStreamingSuite.scala
@@ -70,7 +70,7 @@ class TestCDCStreamingSuite extends HoodieCDCTestBase {
       .option(HoodieTableConfig.CDC_ENABLED.key, "true")
       .option(HoodieTableConfig.CDC_SUPPLEMENTAL_LOGGING_MODE.key, loggingMode.name())
       .option(DataSourceWriteOptions.RECORDKEY_FIELD.key, "userid")
-      .option(DataSourceWriteOptions.PRECOMBINE_FIELD.key, "ts")
+      .option(HoodieTableConfig.ORDERING_FIELDS.key, "ts")
       .option(HoodieWriteConfig.TBL_NAME.key, "user_to_country")
       .save(userToCountryTblPath)
 
@@ -82,7 +82,7 @@ class TestCDCStreamingSuite extends HoodieCDCTestBase {
     countryToPopulationDF.write.format("hudi")
       .options(commonOptions)
       .option(DataSourceWriteOptions.RECORDKEY_FIELD.key, "country")
-      .option(DataSourceWriteOptions.PRECOMBINE_FIELD.key, "ts")
+      .option(HoodieTableConfig.ORDERING_FIELDS.key, "ts")
       .option(HoodieWriteConfig.TBL_NAME.key, "country_to_population")
       .save(countryToPopulationTblPath)
 
@@ -98,7 +98,7 @@ class TestCDCStreamingSuite extends HoodieCDCTestBase {
           .options(commonOptions)
           .option(HoodieTableConfig.CDC_ENABLED.key, "true")
           .option(DataSourceWriteOptions.RECORDKEY_FIELD.key, "userid")
-          .option(DataSourceWriteOptions.PRECOMBINE_FIELD.key, "ts")
+          .option(HoodieTableConfig.ORDERING_FIELDS.key, "ts")
           .option(HoodieWriteConfig.TBL_NAME.key, "user_to_country")
           .mode(SaveMode.Append)
           .save(userToCountryTblPath)
@@ -147,7 +147,7 @@ class TestCDCStreamingSuite extends HoodieCDCTestBase {
           .write.format("hudi")
           .options(commonOptions)
           .option(DataSourceWriteOptions.RECORDKEY_FIELD.key, "country")
-          .option(DataSourceWriteOptions.PRECOMBINE_FIELD.key, "ts")
+          .option(HoodieTableConfig.ORDERING_FIELDS.key, "ts")
           .option(HoodieWriteConfig.TBL_NAME.key, "country_to_population")
           .mode(SaveMode.Append)
           .save(countryToPopulationTblPath)
